Update how-to-ICON_Development_Network-AWS-marketplace.md (#2)
icon-project/documentation: howto/how-to-ICON_Development_Network-AWS-marketplace.md

# How to set up "ICON Development Network" on AWS

## Overview

> This document is a HOW-to document for anyone who wants to set up their own ICON Development Network on AWS.

Reviewer: ```suggestion
This document is for anyone who wants to set up their own ICON Development Network on AWS.
```

## Intended Audience

> Someone who needs their own "ICON Development network" for ICON SCORE or DApp. (Please also indicate level of expertise required)

Reviewer: ```suggestion
Someone who needs their own "ICON Development Network" for ICON SCORE or DApp. (Please also indicate level of expertise required)
```

Reviewer: Need consistency in naming - is it  "ICON Development Network" ?

## Purpose
Following this document step-by-step will all reader to set up their own ICON Development Network on AWS.

## Prerequisites

> ### Create a Key Pair

Reviewer: ```suggestion
### Create an Amazon EC2 Key Pair
```

6. Enter a name for the new key pair in the **Key pair name** field of the **Create Key Pair** dialog box, and then choose **Create**. Use a name that is easy for you to remember, such as your IAM user name, followed by -key-pair, plus the region name. For example, me-key-pair-useast2.
![Key_Pairs_02](./images/Key_Pairs_02.png)
7. The private key file is automatically downloaded by your browser. The base file name is the name you specified as the name of your key pair, and the file name extension is .pem. Save the private key file in a safe place.
8. If you are using an SSH client on a Mac or Linux computer to connect to your Linux instance, use the following command to set the permissions of your private key file so that only you can read it. If you do not set these permissions, then you cannot connect to your instance using this key pair. (Where is the following command?)

### EC2 instance limits

> ** ICON Development network ** requires at least 7 EC2 instances. However, the default limit is 5 for each region, so you need to request extend EC2 instance limitation. You can request from the link below:

Reviewer: ```suggestion
**ICON Development Network** requires at least 7 EC2 instances. However, the default limit is 5 for each region, so you need to request extend EC2 instance limitation. You can request from the link below:
```

 - Node Count: ICON Node Cluster size; must be between 4 and 12. Default value is **4**.
 - InstanceName: EC2 Instance Name Prefix. Default value is **"ICON"**.
 - VpcId: **(Required)** VpcId of your existing Virtual Private Cloud (VPC). Select any drop-down menu item.
 - SubnetID: **(Required)** Subnet should be a public subnet. Select largest Subnet id (for example, \*.\*.0.0/32)
 - Fee: Pay commission to execute the transaction by ICX or not. Default value is **false**
 - Audit: Prevent to deploy SCORE by anybody or not. Default value is **false**
 - KeyName: **(Required)** Name of an existing EC2 KeyPair to enable SSH access to the instance. You must select EC2 Keypair. You must make sure to create a "Keypair" before this setup process.
 - SSHLocation: **(Required)** The IP address range that can be used to SSH to the EC2 instances. The setting with no limitation is "0.0.0.0/0".
![AWS Cloudformation_02](./images/AWS Cloudformation_02.png)
8. Step "Options". Leave as default. Select **Next**
![AWS Cloudformation_03](./images/AWS Cloudformation_03.png)
9. Step "Review". Select **Create**
![AWS Cloudformation_04](./images/AWS Cloudformation_04.png)

> 10. The setup process is over. Just need to check about **Events**.

Reviewer: What kind of information should the engineer look for in the Events tab ? @donghanicon

Reviewer: ```suggestion
10. The setup process is over. Just need to check the **Events** tab below.
```

![AWS Cloudformation_05](./images/AWS Cloudformation_05.png)

> 11. After CloudFormation process is done, check about **Outputs** tab. You can get information for **ICON Development Network**

Reviewer: ```suggestion
11. After the network has been deployed successfully using the CloudFormation, you can get information for **ICON Development Network**  from the **Outputs** tab. 
```

```

###### Send Transaction & Get Transaction Result

> Keystore is needed when sending a transaction or when distributing SCORE. Since it is a development network, try to create it in T-bears, not the actual key file.

Reviewer: ```suggestion
Keystore is needed when sending a transaction or when deploying a SCORE onto the network. You can generate a keystore file using T-Bears CLI.
```
